### PR TITLE
Add checklist-item support

### LINKED ISSUE #116: Retrieve checklist values from Gitlab
Create a `CheckListDirective` which queries GitLab for a Merge Request description and parses the description for checkboxes followed by ITEMID which then determines the pass/fail, checked/unchecked, true/false, x/o value (in below example `checked unchecked`) of the user defined attribute (as `user_defined_attribute`)

Merge request description raw markdown unchecked item:
```markdown
 - [ ] ITEMID This is what we want unchecked
 - [ x ] ITEMID2 This item is checked
```

Which should append attributes to `.. checklist::` to make it
```
.. checklist:: ITEMID Question 1 we want to get answered in Gitlab Merge Request
    :user_defined_attribute: unchecked

.. checklist:: ITEMID2 Question 2 we want to get answered in Gitlab Merge Request
    :user_defined_attribute: checked
```

Then we can generate table of checkboxes with `item-attributes-matrix` inside Sphinx.

### PR DESCRIPTION
**Pull request by JasperCraeghs**
TODO: 
- [x] Update class diagram
- [x] Add this feature to readme
- [x] Make compatible with GitHub (in addition to GitLab)
- [x] Test with GitLab MR
- [x] Test with GitHub PR (public and private)
- [ ] Add example usage of this feature to the example documentation. #121 

Closes #116

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -779,6 +779,76 @@ values of this attribute, ordered in priority from high to low. Using this optio
...
+.. code-block:: bash
+
+    # copy example .env to your .env
+    cp doc/.env.example .env
+
+    # add env variables by adjusting the template values in .env
+
+- *ATTRIBUTE_NAME* is the identifier of the attribute to be added, e.g. *checked*.
+- *ATTRIBUTE_TO_STRING* is the string representation (as to be rendered in html) of the attribute name, e.g. *Answer*.
+- *ATTRIBUTE_VALUES* are two comma-separated attribute values, e.g. *yes/no*. The first value is used when the checkbox is checked and the second value when unchecked.
```
Comment (Letme) [inline README.rst]: if they are comma serparated then use `yes/no` as `yes,no` to avoid confusion

File: README.rst [hunk 2]
```diff
@@ -779,6 +779,76 @@ values of this attribute, ordered in priority from high to low. Using this optio
...
+.. code-block:: rest
+
+    - [x] PLAN-UNIT_TESTS Have you added unit tests for regression detection?
+    - [ ] PLAN-PACKAGE_TEST Have you tested the package?
+
+GitLab
+------
+
+- *PRIVATE_TOKEN* is your personal access token that has API access.
+- *API_HOST_NAME* is the host name of the API, e.g. *https://gitlab.example.com/api/v4*.
```
Comment (Letme) [inline README.rst]: can we remove `api/v4` from contents of this varialbe. `api/v4` is actually something hardcoded in plugin as that is the version of API we will support. We can move higher or lower, but I would not like to have this dynamically inputted by user.

Comment (Letme) [reply README.rst]: Although on the other side if someone has different gitlab instance then maybe this is a preferred way to keep into a separate variable. So nevermind.

File: README.rst (hunk 1 above)
Comment (JasperCraeghs) [reply README.rst]: 68fab7125c5a129aeccf9e44d45eb21a64726e2d

**Pull request merged**